nvflare/private/fed/server/job_runner.py

```diff
@@ -470,6 +477,24 @@ def restore_running_job(self, run_number: str, job_id: str, job_clients, snapsho
                 fl_ctx, f"Failed to restore the job: {job_id} to the running job table: {secure_format_exception(e)}."
             )
 
+    def update_unfinished_jobs(self, fl_ctx: FLContext):
+        engine = fl_ctx.get_engine()
+        job_manager = engine.get_component(SystemComponents.JOB_MANAGER)
+        all_jobs = []
+        dispatched_jobs = job_manager.get_jobs_by_status(RunStatus.DISPATCHED, fl_ctx)
+        all_jobs.extend(dispatched_jobs)
+        running_jobs = job_manager.get_jobs_by_status(RunStatus.RUNNING, fl_ctx)
+        all_jobs.extend(running_jobs)
+
+        for job in all_jobs:
+            try:
+                job_manager.set_status(job.job_id, RunStatus.ABANDONED, fl_ctx)
+                self.logger.info(f"Update the previous running job: {job.job_id} to FAILED_TO_FINISH.")
```

Comment:
suggest change to

`self.log_info(fl_ctx, f"Update the previous running job: {job.job_id} to {RunStatus.ABANDONED}.")`

Reply:
changed.